javascript-source/commands/customCommands.js

```diff
@@ -301,13 +307,19 @@
             message = $.replace(message, '(pay)', String($.inidb.exists('paycom', event.getCommand()) ? $.getPointsString($.inidb.get('paycom', event.getCommand())) : $.getPointsString(0)));
         }
 
-        if (message.match(/\(#\)/g)) {
-            message = $.replace(message, '(#)', String($.randRange(1, 100)));
+        if (message.match(/\(#\)/)) {
+            message = $.replaceFunc(s, '(#)', function () {
+                return String($.randRange(1, 100));
+            });
         }
 
-        if (message.match(/\(# (\d+),(\d+)\)/g)) {
-            var mat = message.match(/\(# (\d+),(\d+)\)/);
-            message = $.replace(message, mat[0], String($.randRange(parseInt(mat[1]), parseInt(mat[2]))));
+        if (message.match(/\(# (\d+),(\d+)\)/)) {
+            var mat = message.match(/\(# (\d+),(\d+)\)/),
+                min = parseInt(mat[1]),
+                max = parseInt(mat[2]);
+            message = $.replaceFunc(message, mat[0], function () {
+                return String($.randRange(min, max));
+            });
```

Comment:
I think this will fail for messages that contain multiple `#` tags with different parameters, like `You open chest number (# 1,10), revealing (# 100,500) gold coins`: only the `(# 1,10)` will be replaced.

(Same goes for the `readfilerand` case above, I think. It'll handle multiple reads of the same file, but not different files.)

Reply:
Indeed. I had another version where that kind of issue would not have happened, but I did some optimization and broke it by doing so :/
Thanks for checking and letting me know.

Reply:
Looking at the code base it seems that all custom command variables with argument(s) have that exact problem. @gmt2001, is this designed like this on purpose?

Reply:
Was for performance in the past

Reply:
"In the past" meas I can be changed or should it stay as is?

Reply:
If you can make it better, go for it